File: core/src/com/broll/gainea/server/core/goals/impl/e1/G_AllDeserts.java
package com.broll.gainea.server.core.goals.impl.e1;

import com.broll.gainea.server.core.goals.OccupyGoal;
import com.broll.gainea.server.core.goals.GoalDifficulty;
import com.broll.gainea.server.core.map.AreaType;
import com.broll.gainea.server.core.map.impl.GaineaMap;

public class G_AllDeserts extends OccupyGoal {
    public G_AllDeserts() {
        super(GoalDifficulty.MEDIUM, "Erobere alle Wüsten");
    }

    @Override
    protected void initOccupations() {
        occupy(area -> area.getType() == AreaType.DESERT, GaineaMap.Areas.values());
    }

}

File: server/src/main/com/broll/gainea/server/core/goals/impl/e1/G_AllDeserts.kt
package com.broll.gainea.server.core.goals.impl.e1

import com.broll.gainea.server.core.goals.GoalDifficulty
import com.broll.gainea.server.core.goals.OccupyGoal
import com.broll.gainea.server.core.map.AreaType
import com.broll.gainea.server.core.map.impl.GaineaMap

class G_AllDeserts : OccupyGoal(GoalDifficulty.MEDIUM, "Erobere alle Wüsten") {
    override fun initOccupations() {
        occupy({ it.type == AreaType.DESERT }, *GaineaMap.Areas.entries.toTypedArray())
    }
}
